### PR TITLE
DPL Analysis: adapt the internals to Configurable<vector>

### PR DESCRIPTION
**Pull request by aalkin**
@jgrosseo @ktf  This is the next step before the "labeled 1/2d-array" configurable can be implemented. It has some string parsing, essentially a hardcoded schema for storing a typed arrays in JSON.

### CODE REVIEW

File: Framework/Core/include/Framework/ConfigParamRegistry.h [hunk 1]
```diff
@@ -12,6 +12,7 @@
 
 #include "Framework/ParamRetriever.h"
 #include "Framework/ConfigParamStore.h"
+#include "Framework/Traits.h"
```
Comment (ktf) [inline Framework/Core/include/Framework/ConfigParamRegistry.h:15]: What do you need from Traits?

Comment (aalkin) [reply Framework/Core/include/Framework/ConfigParamRegistry.h:15]: `is_base_of_template` to check for a `vector<T>`

**Comment by ktf** [comment]
Please, let's wait until later on friday for this. I need to check merging this does not screw up some unrelated development.

**Comment by ktf** [comment]
Ok, this seems fine. It would be nice to have a few more tests for the vector cases, including an empty vector.

**Comment by ktf** [comment]
Regarding the implementation, we should probably either use JSON or msgpack to serialise configuration being reported by the driver to the driver.
